File: TodayNew-OC/TodayNew-OC/CustomTabbarController.m
//
//  CustomTabbarController.m
//  TodayNew-OC
//
//  Created by LXN on 2019/3/21.
//  Copyright © 2019年 LXN. All rights reserved.
//

#import "CustomTabbarController.h"
#import "CustomBar.h"
#import "BaseNavigationController.h"
#import "FirstViewController.h"
#import "SecondViewController.h"
#import "ThirdViewController.h"
#import "FourViewController.h"
#import "CustomBar.h"
@interface CustomTabbarController ()

@end

@implementation CustomTabbarController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    [self addChildViewController];
    self.tabBar.tintColor = [UIColor colorWithRed:245/255.0 green:90/255.0 blue:93/255.0 alpha:1.0];
    [self setValue:[[CustomBar alloc]init] forKey:@"tabBar"];
    
}
- (void)addChildViewController{
    [self setChildControler:[[FirstViewController alloc]init] Title:@"首页" image:@"home_tabbar_32x32_" selectImage:@"home_tabbar_press_32x32_"];
    [self setChildControler:[[SecondViewController alloc]init] Title:@"视频" image:@"weitoutiao_tabbar_32x32_" selectImage:@"weitoutiao_tabbar_press_32x32_"];
    [self setChildControler:[[ThirdViewController alloc]init] Title:@"小视频" image:@"huoshan_tabbar_32x32_" selectImage:@"huoshan_tabbar_press_32x32_"];
    [self setChildControler:[[FourViewController alloc]init] Title:@"我的" image:@"mine_tabbar_32x32_" selectImage:@"mine_tabbar_press_32x32_"];
}
- (void)setChildControler:(UIViewController *)viewController Title:(NSString *)title image:(NSString *)imageName selectImage:(NSString *)selectImage{
    viewController.title = title;
    viewController.tabBarItem.image = [UIImage imageNamed:imageName];
    viewController.tabBarItem.selectedImage = [UIImage imageNamed:selectImage];
    BaseNavigationController *nav = [[BaseNavigationController alloc]initWithRootViewController:viewController];
    
    [self addChildViewController:nav];
    
}
/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: TodayNews/TodayNews/CustomTabbarController.swift
//
//  CustomTabbarController.swift
//  TodayNews
//
//  Created by LXN on 2019/3/21.
//  Copyright © 2019年 LXN. All rights reserved.
//

import UIKit

class CustomTabbarController: UITabBarController {

    override func viewDidLoad() {
        super.viewDidLoad()
        let tabbar = UITabBar.appearance()
        tabbar.tintColor = UIColor(red: 245/255.0, green: 90/255.0, blue: 93/255.0, alpha: 1.0)
        
        setValue(CustomTabbar(), forKey: "tabBar")
        addChildViewController()
    }
    func addChildViewController() {
        setChildViewController(childViewController: FirstViewController(), title: "首页", imageName: "home_tabbar_32x32_", selectImageName: "home_tabbar_press_32x32_")
        
        setChildViewController(childViewController: ThirdViewController(), title: "微头条", imageName: "weitoutiao_tabbar_32x32_", selectImageName: "weitoutiao_tabbar_press_32x32_")
        setChildViewController(childViewController: FourViewController(), title: "小视频", imageName: "huoshan_tabbar_32x32_", selectImageName: "huoshan_tabbar_press_32x32_")
        setChildViewController(childViewController: SecondViewController(), title: "我的", imageName: "mine_tabbar_32x32_", selectImageName: "mine_tabbar_press_32x32_")
        
    }
    func setChildViewController(childViewController:UIViewController,title:String,imageName:String,selectImageName:String){
        childViewController.tabBarItem.image = UIImage(named: imageName)
        childViewController.tabBarItem.selectedImage = UIImage(named: selectImageName)
        childViewController.title = title
        let nav = BaseNavViewController.init(rootViewController: childViewController)
        addChild(nav)
    }

}
